lib/install/realize-shrinkwrap-specifier.js

```diff
@@ -1,14 +1,15 @@
 'use strict'
 var realizePackageSpecifier = require('realize-package-specifier')
 var isRegistrySpecifier = require('./is-registry-specifier.js')
+var url = require('url')
 
 module.exports = function (name, sw, where, cb) {
   function lookup (ver, cb) {
     realizePackageSpecifier(name + '@' + ver, where, cb)
   }
   if (sw.resolved) {
     return lookup(sw.resolved, cb)
-  } else if (sw.from) {
+  } else if (sw.from && url.parse(sw.from).protocol) {
```

Comment:
Maybe pass this to `npa` and if it's not a version or range then do this code path?